website/project/model.py

```diff
@@ -148,13 +147,31 @@ class MetaData(GuidStoredObject):
     date_created = fields.DateTimeField(auto_now_add=datetime.datetime.utcnow)
     date_modified = fields.DateTimeField(auto_now=datetime.datetime.utcnow)
 
+class Commentable(object):
 
-class Comment(GuidStoredObject, SpamMixin):
+    @property
+    def target_type(self):
+        raise NotImplementedError
+
+    @property
+    def root_target_page(self):
```

Comment:
Docstrings for these properties would also be helpful. It isn't obvious to me what `root_target_page` is supposed to return.